swift/extractor/SwiftExtractor.cpp

```diff
@@ -65,25 +66,119 @@ static std::string getFilename(swift::ModuleDecl& module, swift::SourceFile* pri
     filename += module.getName().str();
     return filename;
   }
+  if (module.isBuiltinModule()) {
+    // The Builtin module has an empty filename, let's fix that
+    return "/<Builtin>";
```

Comment:
Drive-by comment: Probably not of much concern right now, but this would be an illegal file name on Windows due to `<` and `>` (see [here](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions)). In practise this might also give some usability problems on other platforms, due to `<` and `>` begin used for redirected input and output (that's assuming the user starts digging into the generated trap files)..

Reply:
I think this is a very good point: right now you can download an existing database and run analysis on Windows, if the db was not cleaned, then it might be problematic for Windows users.
This is of course theoretical, but I think it's worth fixing this.

Reply:
good point! I got inspiration by another module (the mysterious `__Objc`) that actually reports `<imports>` as a filename, leading to `<imports>.trap`. So with the same thing in mind, we might want to also fix that path, right?